.github/workflows/main.yml

```diff
@@ -129,6 +170,7 @@ jobs:
           production-registry-password: ${{ secrets.INFRA_ROOT_AWS_SECRET_ACCESS_KEY }}
 
       - name: build and push dbt server images
+        if: github.ref_name == matrix.dbt-core.server-branch
```

Comment:
Are we missing matrix.dbt-core.server-branch == github.event.pull_request.base.ref here? I feel like both push and pull should trigger update images?

Reply:
No, we only want `push` to trigger an image build and push, not a PR. [This line ](https://github.com/dbt-labs/dbt-server/pull/176/files/82af53e2991e729d93fb3d5583c451006114f539#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3L144)acts as the conditional in this block

Reply:
Please correct me if I'm wrong, shall we want to just test building the image but not push? My guess is that 
- for pull request we should build the image test dockerfile is correct but not push the image.
- for push request we should both build and push?
- The line you pointed out I guess is just to control whether push to ecr. WDYT?

Reply:
That makes sense to me! I wish the actions for build and push were separate in that case, I don't love that it looks from the actions view like we're pushing images on PR, but you're correct that that is how it currently works on main, so I'll keep it the same 👍